ultimarc/devices/_structures.py

```diff
@@ -0,0 +1,90 @@
+#
+# This file is subject to the terms and conditions defined in the
+# file 'LICENSE', which is part of this source code package.
+#
+
+import ctypes as ct
+import logging
+
+_logger = logging.getLogger('ultimarc')
+
+
+class UltimarcStruct(ct.Structure):
+    """ Parent structure """
+    level = 1
+
+    def __repr__(self) -> str:
+        values = ',\n'.join(('  ' * self.level) + f'{name}={value}'
+                            for name, value in self._as_dict_().items())
+        return f'<{self.__class__.__name__}:\n{values}>'
+
+    def _as_dict_(self) -> dict:
+        return {field[0]: (hex(getattr(self, field[0]))
```

Comment:
This method should probably be public instead of private. I could see code that is working with a Structure object wanting to call this.

Reply:
This isn't a true dictionary output any more because of the array inside the structure.  This and the _print_array_  def are used to pretty print the data for debugging through the __repr__ def.  print(structure) would access these functions.

Reply:
Sounds good.